Fix push-packages-and-publish-release step

### Description

https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/21200039490/job/60983685565

```
Run Import-Module .\build\scripts\post-release.psm1
  Import-Module .\build\scripts\post-release.psm1
  
  $HasToken = -Not [string]::IsNullOrEmpty($env:NUGET_TOKEN)
  PushPackagesPublishReleaseUnlockAndPostNoticeOnPrepareReleasePullRequest `
    -gitRepository ${env:GITHUB_REPOSITORY} `
    -pullRequestNumber ${env:ISSUE_NUMBER} `
    -expectedPrAuthorUserName ${env:EXPECTED_PR_AUTHOR_USER_NAME} `
    -commentUserName ${env:COMMENT_USER_NAME} `
    -artifactDownloadPath "${env:GITHUB_WORKSPACE}/artifacts" `
    -pushToNuget $HasToken
  shell: /usr/bin/pwsh -command ". '{0}'"
  env:
    DOTNET_ROOT: /usr/share/dotnet
    GH_TOKEN: ***
    EXPECTED_PR_AUTHOR_USER_NAME: app/otelbot-dotnet
    COMMENT_USER_NAME: Kielek
    ISSUE_NUMBER: 6839
    NUGET_TOKEN: ***
Exception: Could not find package push comment on pr
Error: Process completed with exit code 1.
```

--- a/.github/workflows/automation.yml
+++ b/.github/workflows/automation.yml
@@ -11,6 +11,8 @@ on:
         value: ${{ vars.AUTOMATION_EMAIL }}
       application-name:
         value: ${{ vars.AUTOMATION_APPLICATION_NAME }}
+      application-login:
+        value: ${{ vars.AUTOMATION_APPLICATION_LOGIN }}
       application-username:
         value: ${{ vars.AUTOMATION_APPLICATION_USERNAME }}
     secrets:

--- a/.github/workflows/post-release.yml
+++ b/.github/workflows/post-release.yml
@@ -70,6 +70,7 @@ jobs:
       env:
         GH_TOKEN: ${{ steps.otelbot-token.outputs.token }}
         EXPECTED_PR_AUTHOR_USER_NAME: ${{ needs.automation.outputs.application-name }}
+        EXPECTED_COMMENT_AUTHOR_USER_NAME: ${{ needs.automation.outputs.application-login }}
         COMMENT_USER_NAME: ${{ github.event.comment.user.login }}
         ISSUE_NUMBER: ${{ github.event.issue.number }}
         NUGET_TOKEN: ${{ steps.nuget-login.outputs.NUGET_API_KEY }}
@@ -81,6 +82,7 @@ jobs:
           -gitRepository ${env:GITHUB_REPOSITORY} `
           -pullRequestNumber ${env:ISSUE_NUMBER} `
           -expectedPrAuthorUserName ${env:EXPECTED_PR_AUTHOR_USER_NAME} `
+          -expectedCommentAuthorUserName ${env:EXPECTED_COMMENT_AUTHOR_USER_NAME} `
           -commentUserName ${env:COMMENT_USER_NAME} `
           -artifactDownloadPath "${env:GITHUB_WORKSPACE}/artifacts" `
           -pushToNuget $HasToken
@@ -153,6 +155,7 @@ jobs:
       env:
         GH_TOKEN: ${{ steps.otelbot-token.outputs.token }}
         EXPECTED_PR_AUTHOR_USER_NAME: ${{ needs.automation.outputs.application-name }}
+        EXPECTED_COMMENT_AUTHOR_USER_NAME: ${{ needs.automation.outputs.application-login }}
         TAG: ${{ inputs.tag || github.ref_name }}
       run: |
         Import-Module .\build\scripts\post-release.psm1
@@ -160,4 +163,5 @@ jobs:
         TryPostReleasePublishedNoticeOnPrepareReleasePullRequest `
           -gitRepository ${env:GITHUB_REPOSITORY} `
           -expectedPrAuthorUserName ${env:EXPECTED_PR_AUTHOR_USER_NAME} `
+          -expectedCommentAuthorUserName ${env:EXPECTED_COMMENT_AUTHOR_USER_NAME} `
           -tag ${env:TAG}

--- a/.github/workflows/publish-packages-1.0.yml
+++ b/.github/workflows/publish-packages-1.0.yml
@@ -227,6 +227,7 @@ jobs:
       env:
         GH_TOKEN: ${{ steps.otelbot-token.outputs.token }}
         EXPECTED_PR_AUTHOR_USER_NAME: ${{ needs.automation.outputs.application-name }}
+        EXPECTED_COMMENT_AUTHOR_USER_NAME: ${{ needs.automation.outputs.application-login }}
         PACKAGES_URL: ${{ needs.build-pack-publish.outputs.artifact-url }}
       run: |
         Import-Module .\build\scripts\post-release.psm1
@@ -236,4 +237,5 @@ jobs:
           -tag ${env:GITHUB_REF_NAME} `
           -tagSha ${env:GITHUB_SHA} `
           -packagesUrl ${env:PACKAGES_URL} `
-          -expectedPrAuthorUserName ${env:EXPECTED_PR_AUTHOR_USER_NAME}
+          -expectedPrAuthorUserName ${env:EXPECTED_PR_AUTHOR_USER_NAME} `
+          -expectedCommentAuthorUserName ${env:EXPECTED_COMMENT_AUTHOR_USER_NAME}

--- a/build/scripts/post-release.psm1
+++ b/build/scripts/post-release.psm1
@@ -125,7 +125,8 @@ function TryPostPackagesReadyNoticeOnPrepareReleasePullRequest {
     [Parameter(Mandatory=$true)][string]$tag,
     [Parameter(Mandatory=$true)][string]$tagSha,
     [Parameter(Mandatory=$true)][string]$packagesUrl,
-    [Parameter(Mandatory=$true)][string]$expectedPrAuthorUserName
+    [Parameter(Mandatory=$true)][string]$expectedPrAuthorUserName,
+    [Parameter(Mandatory=$true)][string]$expectedCommentAuthorUserName
   )
 
   $prListResponse = gh pr list --search $tagSha --state merged --json number,author,title,comments | ConvertFrom-Json
@@ -146,7 +147,7 @@ function TryPostPackagesReadyNoticeOnPrepareReleasePullRequest {
     $foundComment = $false
     foreach ($comment in $pr.comments)
     {
-      if ($comment.author.login -eq $expectedPrAuthorUserName -and $comment.body.StartsWith("I just pushed the [$tag]"))
+      if ($comment.author.login -eq $expectedCommentAuthorUserName -and $comment.body.StartsWith("I just pushed the [$tag]"))
       {
         $foundComment = $true
         break
@@ -181,6 +182,7 @@ function PushPackagesPublishReleaseUnlockAndPostNoticeOnPrepareReleasePullReques
     [Parameter(Mandatory=$true)][string]$gitRepository,
     [Parameter(Mandatory=$true)][string]$pullRequestNumber,
     [Parameter(Mandatory=$true)][string]$expectedPrAuthorUserName,
+    [Parameter(Mandatory=$true)][string]$expectedCommentAuthorUserName,
     [Parameter(Mandatory=$true)][string]$commentUserName,
     [Parameter(Mandatory=$true)][string]$artifactDownloadPath,
     [Parameter(Mandatory=$true)][bool]$pushToNuget
@@ -215,7 +217,7 @@ function PushPackagesPublishReleaseUnlockAndPostNoticeOnPrepareReleasePullReques
   $packagesUrl = ''
   foreach ($comment in $prViewResponse.comments)
   {
-    if ($comment.author.login -eq $expectedPrAuthorUserName -and $comment.body.StartsWith("The packages for [$tag](https://github.com/$gitRepository/releases/tag/$tag) are now available:"))
+    if ($comment.author.login -eq $expectedCommentAuthorUserName -and $comment.body.StartsWith("The packages for [$tag](https://github.com/$gitRepository/releases/tag/$tag) are now available:"))
     {
       $foundComment = $true
       break
@@ -550,6 +552,7 @@ function TryPostReleasePublishedNoticeOnPrepareReleasePullRequest {
   param(
     [Parameter(Mandatory=$true)][string]$gitRepository,
     [Parameter(Mandatory=$true)][string]$expectedPrAuthorUserName,
+    [Parameter(Mandatory=$true)][string]$expectedCommentAuthorUserName,
     [Parameter(Mandatory=$true)][string]$tag
   )
 
@@ -577,7 +580,7 @@ function TryPostReleasePublishedNoticeOnPrepareReleasePullRequest {
     $foundComment = $false
     foreach ($comment in $pr.comments)
     {
-      if ($comment.author.login -eq $expectedPrAuthorUserName -and $comment.body.StartsWith("The packages for [$tag](https://github.com/$gitRepository/releases/tag/$tag) are now available:"))
+      if ($comment.author.login -eq $expectedCommentAuthorUserName -and $comment.body.StartsWith("The packages for [$tag](https://github.com/$gitRepository/releases/tag/$tag) are now available:"))
       {
         $foundComment = $true
         break

